ql/src/java/org/apache/hadoop/hive/ql/DriverTxnHandler.java

```diff
@@ -570,7 +570,7 @@ void endTransactionAndCleanup(boolean commit) throws LockException {
     txnRollbackRunner = null;
   }
 
-  void endTransactionAndCleanup(boolean commit, HiveTxnManager txnManager) throws LockException {
+  synchronized void endTransactionAndCleanup(boolean commit, HiveTxnManager txnManager) throws LockException {
```

Comment:
Could we leave a comment here, why is this synchronized?

Reply:
added